src/qt/scicon.cpp

```diff
@@ -11,6 +11,7 @@
 #include <QPalette>
 #include <QPixmap>
 
+#if !defined(WIN32) && !defined(MAC_OSX)
 static void MakeSingleColorImage(QImage& img, const QColor& colorbase)
 {
     img = img.convertToFormat(QImage::Format_ARGB32);
```

Comment:
The point is not to reduce the *number* of ifdefs, but to require the compiler to compile the largest possible share of the code on all platforms.